core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -301,7 +253,7 @@ sealed trait Message[F[_]] extends Media[F] { self =>
   def toStrict(implicit F: Concurrent[F]): F[Self] =
     body.compile.to(Chunk).map { chunk =>
       withBodyStream(Stream.chunk(chunk))
-        .withContentLength(`Content-Length`.unsafeFromLong(chunk.size.toLong))
+        .transformHeaders(_.withContentLength(`Content-Length`.unsafeFromLong(chunk.size.toLong)))
     }
```

Comment:
Ah sorry, missed this. Hm. Also now I realize I'm not sure how this interacts with https://github.com/http4s/http4s/pull/6286 which places the `Content-Length` header on the entity itself. Shouldn't that be the canonical?

Reply:
Well, that would be introduced on main, and doesn't change this for 0.23 purposes, right?  I don't think #6286 is a reason to hold this after all?

Reply:
Ah, right. Didn't mean to hold this one up.